src/Utility/TestMailer.php

```diff
@@ -27,6 +31,24 @@ public function __construct(
         $this->testSessionEnvironment = TestSessionEnvironment::singleton();
     }
 
+    public function send(RawMessage $message, Envelope $envelope = null): void
+    {
+        parent::send($message, $envelope);
+        if (!is_a($message, Email::class)) {
+            throw new InvalidArgumentException('$message must be a ' . Email::class);
+        }
+        /** @var Email $email */
+        $email = $message;
+        $data = $this->createData($email);
```

Comment:
```suggestion
        $data = end($this->emailsSent);
```
`parent::send()` already created this data for us.

Reply:
Can't do this, $this->emailsSent is private, not protected

Reply:
Fair enough